02stl/cpp/particlesMMap.cc

```diff
@@ -0,0 +1,25 @@
+#include <stdio.h>
+#include <map>
+#include <string>
+#include <iostream>
+
+int main()
+{
+  FILE* ifp = fopen("02stl/cpp/particleList.txt","r");
+
+  // Read in the data
+  std::multimap<std::string,float> theParticles;
+  char name[80];
+  float momentum;
+  while (!feof(ifp)) {
+    fscanf(ifp, "%s %f", name, &momentum);
```

Comment:
Again, that's complicated C syntax. With hard-coded limits on the size of name, rigid file syntax (number of white spaces, etc). Exactly the kind of horror we do NOT want to teach them by default.

File streams and std::string are the answer.